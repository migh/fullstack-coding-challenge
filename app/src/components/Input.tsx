import * as React from 'react';

export interface InputProps {
  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
  onEnter: (event: React.FormEvent<HTMLInputElement>) => void;
  placeholder: string;
  value: string;
}

export const Input = (props: InputProps) => (
    <input
      className="form-control"
      type="text"
      placeholder={props.placeholder}
      value={props.value}
      onChange={props.onChange}
      onKeyPress={props.onEnter}
    />
);

