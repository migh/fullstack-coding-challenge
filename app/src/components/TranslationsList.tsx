import * as React from 'react';
import { Translation } from './Input.container';

// I know lodash have a function for this
function capitalize(str: string): string {
  return str.slice(0,1).toUpperCase() + str.slice(1);
}

// Orders the longer string first
function orderByLength(a: Translation, b: Translation): number {
  const aLength = a.textTarget && a.textTarget.length || 0;
  const bLength = b.textTarget && b.textTarget.length || 0;
  return bLength - aLength;
}

// I know classnames could work for this
function getClass(status: string): string {
  switch(status) {
    case 'requested':
      return 'list-group-item-success';
    case 'pending':
      return 'list-group-item-warning';
    case 'completed':
      return 'list-group-item-primary';
  }
}

export interface TranslationsListProps {
  isLoaded: boolean;
  translations: Translation[];
  onItemClick: (item: Translation) => () => void;
}

export const TranslationsList = (props: TranslationsListProps) => (
    <div className="row justify-content-md-center">
      <div className="col-10">
        <ul className="list-group">
          {props.translations.sort(orderByLength).map(translation => (
            <li
              key={translation.id}
              className={`list-group-item mb-1 ${getClass(translation.status)}`}
              style={{ cursor: 'pointer' }}
              onClick={props.onItemClick(translation)}
            >
              {translation.textSource}
              <span className="float-right">{capitalize(translation.status)}</span>
              {translation.status === 'completed' && (
                <React.Fragment>
                  <hr />
                  <div>{translation.textTarget}</div>
                </React.Fragment>
              )}
            </li>
          ))}
        </ul>
      </div>
    </div>
);

