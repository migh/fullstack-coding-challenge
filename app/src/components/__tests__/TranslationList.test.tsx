import * as React from 'react';
import { shallow } from 'enzyme';

import { Translation } from '../Input.container';
import { TranslationsList } from '../TranslationsList';

describe('<TranslationsList />', () => {
  describe('Snapshots', () => {
    it('should render.', () => {
      const props = {
        isLoaded: true,
        translations: [
          {
            id: 1,
            jobId: '34234',
            textSource: 'Uno',
            textTarget: 'One',
            status: 'completed'
          },
          {
            id: 2,
            jobId: 'dfsgs3',
            textSource: 'Two',
            textTarget: 'Dos',
            status: 'pending'
          }
        ] as Translation[],
        onItemClick: () => () => {}
      };
      expect(shallow(<TranslationsList {...props} />)).toMatchSnapshot();
    });
  });
});
