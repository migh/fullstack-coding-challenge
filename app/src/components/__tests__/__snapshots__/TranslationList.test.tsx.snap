// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TranslationsList /> Snapshots should render. 1`] = `ShallowWrapper {}`;
